client/settings/payment-settings/account-keys-modal.js

```diff
@@ -73,7 +79,10 @@ const SecretKey = () => {
 				'woocommerce-gateway-stripe'
 			) }
 			value={ value }
-			onChange={ ( val ) => setValue( val ) }
+			onChange={ ( val ) => {
+				setValue( val );
+				setSecretKey( val );
```

Comment:
When the user changes their secret, I now call the setter (`setSecretKey`) so other consumers of the `useAccountKeysSecretKey()` hook can get the latest key.